controllers/drplacementcontrol.go

```diff
@@ -535,20 +535,30 @@ func (d *DRPCInstance) checkMetroFailoverPrerequisites(curHomeCluster string) (b
 func (d *DRPCInstance) checkRegionalFailoverPrerequisites() bool {
 	d.setProgression(rmn.ProgressionWaitForStorageMaintenanceActivation)
 
+	drRearrangedClusters := d.drClusters
```

Comment:
This does not copy the slice, so the next loop append the failover cluster to the end of the original list.

I think we want a helper returning the failover cluster from drClusters, that can be used in other places instead of repeating the loop, and create a new slice with it for call.

We also want a comment explaining *why* we do this:

    // Why we want to use only the failover cluster during failover...
    failoverCluster := d.failoverCluster()    
    s3Clusters := []rmn.DRCluster{failoverCluster}